src/tests/multihost/alltests/test_misc.py

```diff
@@ -532,3 +533,57 @@ def test_0009_dbus_method_find_usrby_attr(multihost, backupsssdconf, ldap_posix_
             assert cmd2.stdout_text.strip('\n') in cmd.stdout_text, 'dbus is not fetching expected users'
         cmd1 = multihost.client[0].run_command(f'id -u {usr}@{domain_name}', raiseonerr=False)
         assert cmd1.stdout_text.strip('\n') not in cmd.stdout_text, 'dbus is fetching unwanted user'
+
+    @staticmethod
+    @pytest.mark.tier1_2
+    def test_file_descriptors(multihost):
+        """
+        :title: sssd-be tends to run out of system resources, hitting the maximum number of open files
+        :id: d6fc8420-0119-11ee-a414-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2195919
+        :customerscenario: true
+        :setup:
+          1. Replaced /usr/libexec/sssd/krb5_child with a shell script like
+            #!/bin/bash
+            sleep 10
+        :steps:
+          1. Run authentications from shell
+          2. List no of file descriptors in `ls -al /proc/$(pidof sssd_be)/fd` before modification of krb5_child
+          3. Do 10 more authentications from shell
+          4. List no of file descriptors in `ls -al /proc/$(pidof sssd_be)/fd` after modification of krb5_child
+          5. Compare no of file descriptors before and after modification of krb5_child
+        :expectedresults:
+          1. Authentications happens successfully
+          2. Redirect the output to  /tmp/before_count
+          3. All 10 authentications will fail
+          4. Redirect the output to  /tmp/after_count
+          5. No of file descriptors should be same as before and after modification of krb5_child
+        :teardown:
+          1. Restore krb5_child
+          2. Remove after_count and before_count files
+        """
+        client = multihost.client[0]
+        user = f'foo1@{ds_instance_name}'
+        check_login(user, multihost.client[0].sys_hostname, "Secret123")
+        time.sleep(3)
+        client.run_command('ls -al /proc/$(pidof sssd_be)/fd > /tmp/before_count')
+        time.sleep(2)
+        n_log_bfr = count_pattern_logs(multihost, "/tmp/before_count", "pipe:")
+        client.run_command("cp -vf /usr/libexec/sssd/krb5_child /usr/libexec/sssd/krb5_child_bkp")
+        client.run_command("echo '#!/bin/bash'> /usr/libexec/sssd/krb5_child")
+        client.run_command("echo 'sleep 10' >> /usr/libexec/sssd/krb5_child")
```

Comment:
To help limit/reduce the amount of bash code in our tests, can you use the pytest multihost get_file_contents and put_file_contents methods for copying and writing files?

For the cp, it'd be something like:

```
krb5_child = client.get_file_contents('/usr/libexec/sssd/krb5_child')
client.put_file_contents('/usr/libexec/sssd/krb5_child_bkp', krb5_child)

```
I'm not sure if you need to pass encoding='utf-8' to the get_file_contents since krb5_child is a binary, that may be unnecessary.

For writing the new contents to the file, you can assign value to a variable and then use put_file_contents() to write to it.

```
    new_child = textwrap.dedent("""\
        #!/bin/bash
        sleep 10
    """)
    client.put_file_contents('/usr/libexec/sssd/krb5_child', new_child)
```